source/util/AMScalerCountAnalyser.cpp

```diff
@@ -0,0 +1,145 @@
+#include "AMScalerCountAnalyser.h"
+#include <QStringList>
+
+AMScalerCountAnalyser::AMScalerCountAnalyser(const QVector<double>& scalerCounts,
+                                             double countNoiseThreshold,
+					                         double countChangeThreshold)
+{
+	scalerCounts_ = scalerCounts;
+	countNoiseThreshold_ = countNoiseThreshold;
+	countChangeThreshold_ = countChangeThreshold;
+	locateTransitions();
+	locateMotionPeriods();
+}
+
+QString AMScalerCountAnalyser::toString()
+{
+	QStringList elementsList;
+
+	for(int iRawScalerData = 0, rawDataCount = scalerCounts_.count();
+	    iRawScalerData < rawDataCount;
+	    ++iRawScalerData) {
+
+		elementsList.append(QString("%1\t%2").arg(iRawScalerData).arg(scalerCounts_.at(iRawScalerData)));
+	}
+
+	foreach(int transitionIndex, transitionIndices_) {
+
+		elementsList[transitionIndex].prepend("*");
+	}
+
+	for(int iMotionPeriod = 0, count = motionPeriods_.count();
+	    iMotionPeriod < count;
+	    ++iMotionPeriod) {
+
+		elementsList[motionPeriods_.at(iMotionPeriod).first].append("\tStart");
+		elementsList[motionPeriods_.at(iMotionPeriod).second].append("\tEnd");
+	}
+
+	QString finalString = QString("Index\tValue\n");
+	foreach(QString indexString, elementsList) {
+
+		finalString.append(indexString + "\n");
+	}
+
+	return finalString;
+}
+
+
+QVector<QPair<int, int> > AMScalerCountAnalyser::motionPeriods() const
+{
+	return motionPeriods_;
+}
+
+void AMScalerCountAnalyser::locateTransitions()
+{
+
+	int previousCountDifference = 0;
+
+	for (int iScalerCount = 1, totalCounts = scalerCounts_.count(); iScalerCount < totalCounts; ++iScalerCount) {
+
+		int currentCountDifference = scalerCounts_.at(iScalerCount - 1) - scalerCounts_.at(iScalerCount);
+
+		if(iScalerCount == 1 && scalerCounts_.at(0) > countNoiseThreshold_ ) {
+			// The first element is above noise tolerance, mark it as a transition
+
+			transitionIndices_.append(0);
```

Comment:
How many points are we expecting here?  Appending to QVector's is generally not a great idea.